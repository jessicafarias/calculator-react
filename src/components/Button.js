import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  name, click, wide, color,
}) => (
  <button
    style={{ width: wide ? '50%' : '25%', backgroundColor: color }}
    onClick={() => click(name)}
    key={name.toString()}
    type="button"
  >
    {name}
  </button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
  click: PropTypes.func,
  wide: PropTypes.bool,
  color: PropTypes.string,

};

Button.defaultProps = {
  color: '#F5913E',
  wide: false,
  click: () => {},

};

export default Button;
