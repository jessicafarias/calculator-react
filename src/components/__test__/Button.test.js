import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import Button from '../Button';

describe('Display Home.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<BrowserRouter><Button name="AC" /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
