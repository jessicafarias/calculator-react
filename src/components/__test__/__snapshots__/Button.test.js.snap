// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Home.js renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#F5913E",
      "width": "25%",
    }
  }
  type="button"
>
  AC
</button>
`;
